### PR TITLE
Prep for GQL

### PR DESCRIPTION
**Pull request by richmolj**
Mostly bug fixes/improvements to flat JSON rendering, which some minor edits to respect GraphQL context. Been seen through the individual commits.

### CODE REVIEW

File: lib/graphiti/hash_renderer.rb [hunk 1]
```diff
@@ -1,39 +1,63 @@
...
         rels = @_relationships.select { |k, v| !!include[k] }
         rels.each_with_object({}) do |(k, v), h|
           serializers = v.send(:resources)
-          attrs[k] = if serializers.is_a?(Array)
+          name = graphql ? k.to_s.camelize(:lower) : k
+          name_chain << k unless name_chain.last == k
+          attrs[name.to_sym] = if serializers.is_a?(Array)
             serializers.map do |rr| # use private method to avoid array casting
-              rr.to_hash(fields: fields, include: include[k])
+              rr.to_hash(fields: fields, include: include[k], graphql: graphql, name_chain: name_chain)
```
Comment (wadetandy) [inline lib/graphiti/hash_renderer.rb:39]: Should this be `name_chain.dup` to avoid accidentally modifying the same thing across multiple loops?

**Review by wadetandy — APPROVED** [review]
One comment/question, but this looks 💯

File: lib/graphiti/hash_renderer.rb (hunk 1 above)
Comment (richmolj) [reply lib/graphiti/hash_renderer.rb:39]: Will work on adding a spec for this in either case, good thought

**Comment by richmolj** [comment]
@wadetandy added spec and `.dup` - some very minor performance impact in edge cases

**Pull request merged**